src/map/mob.cpp

```diff
@@ -1917,8 +1911,21 @@ static bool mob_ai_sub_hard(struct mob_data *md, t_tick tick)
 		return true;
 	}
 
-	//Attempt to attack.
-	//At this point we know the target is attackable, we just gotta check if the range matches.
+	// At this point we know the target is attackable, attempt to attack
+
+	// Monsters in angry state, after having used a normal attack, will always attempt a skill
+	if (md->ud.walktimer == INVALID_TIMER && md->state.skillstate == MSS_ANGRY && md->ud.skill_id == 0)
```

Comment:
This function is called every 100ms, so it would also be called while the monster is moving, though I'm not sure why it's state would still be MSS_ANGRY if it does. The check was already there before so I probably put it there for a reason at some point years ago. Need to be careful changing stuff here, because you often think something is not needed until you realize what you broke by removing it.

Reply:
The `md->ud.walktimer == INVALID_TIMER` part was already there. Not a big fan of leaving TODOs in the source code. I go by the "Don't fix a working system".